File: arc_prediction/verify_rpcs.js
const { ethers } = require('ethers');

const rpcs = [
    "https://5042002.rpc.thirdweb.com",
    "https://rpc.testnet.arc.network"
];

async function run() {
    for (const rpc of rpcs) {
        console.log(`\nChecking ${rpc}...`);
        try {
            const provider = new ethers.JsonRpcProvider(rpc);
            const net = await provider.getNetwork();
            console.log(`Chain ID: ${net.chainId.toString()}`);
            const blockNum = await provider.getBlockNumber();
            console.log(`Block Number: ${blockNum}`);
        } catch (e) {
            console.log(`Error: ${e.message}`);
        }
    }
}

run();

File: tests_and_debugging/rpc/verify_rpcs.py
import requests

rpcs = [
    "https://arc-testnet.g.alchemy.com/v2/gmklUsP-qeITLeu6a8Pw1",
    "https://rpc.testnet.arc.network"
]

for rpc in rpcs:
    print(f"\nChecking {rpc}...")
    try:
        response = requests.post(rpc, json={"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":1}, timeout=10)
        print(f"Chain ID: {response.json().get('result')}")
        
        # Also check block number to see if it's lagging
        response = requests.post(rpc, json={"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1}, timeout=10)
        print(f"Block Number: {response.json().get('result')}")
    except Exception as e:
        print(f"Error: {e}")
